src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.cpp

```diff
@@ -0,0 +1,645 @@
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+}
+
+void PushAvStreamTransportServer::HandleManuallyTriggerTransport(
+    HandlerContext & ctx, const Commands::ManuallyTriggerTransport::DecodableType & commandData)
+{
+    Status status           = Status::Success;
+    uint16_t connectionID   = commandData.connectionID;
+    auto & activationReason = commandData.activationReason;
+    auto & timeControl      = commandData.timeControl;
+
+    TransportConfigurationStorageWithFabricIndex * transportConfiguration = FindStreamTransportConnection(connectionID);
+
+    if (transportConfiguration == nullptr)
+    {
+        ChipLogError(Zcl, "HandleManuallyTriggerTransport: ConnectionID Not Found.");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+        return;
+    }
+
+    if (transportConfiguration->fabricIndex != ctx.mCommandHandler.GetAccessingFabricIndex())
+    {
+        ChipLogError(Zcl, "HandleManuallyTriggerTransport: ConnectionID Not Found.");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+        return;
+    }
+
+    if (mDelegate.GetTransportStatus(connectionID) == PushAvStreamTransportStatusEnum::kBusy)
+    {
+        ChipLogError(Zcl, "HandleManuallyTriggerTransport: Connection is Busy");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Busy);
+        return;
+    }
+
+    if (transportConfiguration->transportConfiguration.transportStatus == TransportStatusEnum::kInactive)
+    {
+        auto clusterStatus = static_cast<uint8_t>(StatusCodeEnum::kInvalidTransportStatus);
+        ChipLogError(Zcl, "HandleManuallyTriggerTransport: Invalid Transport status");
+        ctx.mCommandHandler.AddClusterSpecificFailure(ctx.mRequestPath, clusterStatus);
+        return;
+    }
+    if (transportConfiguration->transportConfiguration.transportOptions.HasValue())
+    {
+        if (transportConfiguration->transportConfiguration.transportOptions.Value().triggerOptions.triggerType ==
+            TransportTriggerTypeEnum::kContinuous)
+        {
+
+            auto clusterStatus = static_cast<uint8_t>(StatusCodeEnum::kInvalidTriggerType);
+            ChipLogError(Zcl, "HandleManuallyTriggerTransport: Invalid Trigger type");
+            ctx.mCommandHandler.AddClusterSpecificFailure(ctx.mRequestPath, clusterStatus);
+            return;
+        }
+        if (transportConfiguration->transportConfiguration.transportOptions.Value().triggerOptions.triggerType ==
+                TransportTriggerTypeEnum::kCommand &&
+            !timeControl.HasValue())
+        {
+
+            ChipLogError(Zcl, "HandleManuallyTriggerTransport: Time control field not present");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+            return;
+        }
+    }
+
+    // Call the delegate
+    status = mDelegate.ManuallyTriggerTransport(connectionID, activationReason, timeControl);
```

Comment:
So the handling of the "TimeControl field is not present" bits of the spec needs to be handled by the delegate?  That needs to be clearly documented.

In general, it would be best if the delegate API implementation made it clear exactly what parts of the spec the delegate is responsible for, vs what's handled by the cluster.